qiskit/circuit/quantumcircuit.py

```diff
@@ -1756,22 +1756,29 @@ def draw(
             cregbundle=cregbundle,
         )
 
-    def size(self) -> int:
-        """Returns total number of gate operations in circuit.
+    def size(self, filter_function: Optional[callable] = lambda x: not x[0]._directive) -> int:
+        """Returns total number of instructions in circuit.
+
+        Args:
+            filter_function (callable): a function to filter out some instructions.
+                Should take as input a tuple of (Instruction, list(Qubit), list(Clbit)).
```

Comment:
I'm wondering if it makes more sense to flatten this one layer and use 3 positional args. It might be easier for people to create functions like:

```python
def filter_fn(instruction, qargs, cargs):
```

instead of:
```python3
def filter_fn(inst_tuple):
    instruction = inst_tuple[0]
    qargs =  inst_tuple[1]
    cargs =  inst_tuple[2]
```

Reply:
I intentionally did it this way because I think we really need something in Qiskit to represent the tuple (inst, qubit, clbits). I was actually going to do that as a followup, using a namedtuple or something. It comes up over and over again that you need the operation plus the operands. I think that is the basic unit of a circuit (i.e. one line of qasm).